code/game/objects/items/cigs_lighters.dm

```diff
@@ -542,26 +545,18 @@ CIGARETTE PACKETS ARE IN FANCY.DM
 	STOP_PROCESSING(SSobj, src)
 	. = ..()
 
-/obj/item/clothing/mask/cigarette/pipe/process(delta_time)
-	var/turf/location = get_turf(src)
-	smoketime -= delta_time
-	if(smoketime <= 0)
-		new /obj/effect/decal/cleanable/ash(location)
-		if(ismob(loc))
-			var/mob/living/M = loc
-			to_chat(M, "<span class='notice'>Your [name] goes out.</span>")
-			lit = FALSE
-			icon_state = icon_off
-			inhand_icon_state = icon_off
-			M.update_inv_wear_mask()
-			packeditem = FALSE
-			name = "empty [initial(name)]"
-		STOP_PROCESSING(SSobj, src)
-		return
-	open_flame()
-	if(reagents?.total_volume) // check if it has any reagents at all
-		handle_reagents()
-
+/obj/item/clothing/mask/cigarette/pipe/go_out(turf/location, mob/living/M)
+	new /obj/effect/decal/cleanable/ash(location)
+	if(ismob(M))
+		to_chat(M, "<span class='notice'>Your [name] goes out.</span>")
+		lit = FALSE
+		icon_state = icon_off
+		inhand_icon_state = icon_off
+		M.update_inv_wear_mask()
+		packeditem = FALSE
+		name = "empty [initial(name)]"
+	STOP_PROCESSING(SSobj, src)
```

Comment:
You removed `process` itself. This should no longer be necessary, and you should remove all other references to processing, right?

Reply:
actually this is because if you remove this, it will act like the pipe is still being smoked when it isnt